java-frontend/src/test/java/org/sonar/java/model/ExpressionUtilsTest.java

```diff
@@ -284,6 +285,23 @@ void enclosing_method_test() {
     });
   }
 
+  @Test
+  void parent_of_type_method_test() {
+    File file = new File("src/test/files/model/ExpressionEnclosingMethodTest.java");
+    CompilationUnitTree tree = JParserTestUtils.parse(file);
+    FindAssignment findAssignment = new FindAssignment();
+    tree.accept(findAssignment);
+    findAssignment.assignments.forEach(a -> {
+      ClassTree parent = (ClassTree) ExpressionUtils.getParentOfType(a, Tree.Kind.CLASS, Tree.Kind.INTERFACE);
+      if (parent.is(Tree.Kind.CLASS)) {
+        assertThat(parent.simpleName().name()).isEqualTo("Clazz");
+      }
+      if (parent.is(Tree.Kind.INTERFACE)) {
```

Comment:
Can a tree be a class and interface? Otherwise, I would join these two ifs as mutually exclusive branches of the same tree.
```suggestion
      } else if (parent.is(Tree.Kind.INTERFACE)) {
```